cocos/scene-graph/node.ts

```diff
@@ -2101,41 +2119,54 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
         let childMat: Mat4;
         let childPos: Vec3;
         let dirtyBits = 0;
+        let positionDirty = 0;
+        let rotationScaleDirty = 0;
 
         while (i) {
             child = dirtyNodes[--i];
             childMat = child._mat;
             childPos = child._pos;
             dirtyBits |= child._transformFlags;
+            positionDirty = dirtyBits & TransformBit.POSITION;
+            rotationScaleDirty = dirtyBits & TransformBit.RS;
             if (cur) {
-                if (dirtyBits & TransformBit.POSITION) {
+                if (positionDirty && !rotationScaleDirty) {
                     Vec3.transformMat4(childPos, child._lpos, cur._mat);
```

Comment:
If position and rotation scale are all dirty. Doesn't need to go here to calculate world matrix since in the following `if (rotationScaleDirty) {` logic , it will do that in `Mat4.fromSRT ( get the local matrix)` + `Mat4.multiply ( get the world matrix = parent's world matrix * my local matrix )`.

In this case, we avoid the unused `Vec3.transformMat4` operation.